toolkit/tools/pkg/simplechroottool/simplechroottool.go

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// A common interface for creating a basic spec/rpm parsing environment inside a chroot.
+
+package simplechroottool
+
+import (
+	"path/filepath"
+	"runtime"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/logger"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/rpm"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/safechroot"
+)
+
+const (
+	chrootSpecDirPath = "/SPECS"
+)
+
+// SimpleChrootTool is a tool for creating a basic spec/rpm parsing environment inside a chroot. When creating a new instance, the
+// following fields must be set:
+//   - BuildDirPath: The path to the directory where the chroot will be created and run from.
+//   - WorkerTarPath: The path to the tarball containing the chroot environment.
+// The following fields will be set by InitializeChroot():
+//   - chroot: The chroot environment we are working in, call Chroot.Run() to execute commands inside the chroot.
+//   - chrootSpecDir: The path to the SPECS directory inside the chroot "/SPECS."
+
+type SimpleChrootTool struct {
```

Comment:
From offline discussion: think what we want to achieve with the new tool to not make it so generic, that it is duplicate of `safechroot` but also not limit its purpose to narrowly.

One idea you mentioned: separate the `BuildCompatibleSpecsList` function into a separate module and make this become just a way to define a more robust chroot environment with pre-defined mount points and directories for general use.

Suggestion: in this case it might be possible that the default defines may need to moved out along with `BuildCompatibleSpecsList`, since they will no longer make sense for a general-purpose chroot.

Reply:
I'm thinking that part of this well-defined chroot is the `distTag` and `runChecks` state. I'm thinking we leave the default defines as part of the chroot wrapper package since it needs to track those.